crates/sui-types/src/messages.rs

```diff
@@ -96,6 +96,29 @@ pub struct TransferSui {
     pub amount: Option<u64>,
 }
 
+/// Gas will be paid from input coins, the remaining coins will
+/// all be paid to recipient.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct PayAllSui {
```

Comment:
Is the plan to eventually deprecate TransferSui? If so, please add a TODO above TransferSui

Reply:
I prefer to eventually deprecate TransferSui if that makes sense, will add the TODO here.